src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,871 @@
+        self._speed_steps = {}  # axis name (str) -> int, speed in steps per meter
+        self._portpattern = port
+
+        # Conversion factors
+        # Count refers to encoder counts, step refers to motor steps. The encoder counts are fixed and given
+        # as a parameter to the axis, the motor counts are determined during configuration and are usually
+        # stored in flash memory.
+        # ._steps_per_meter is redundant, but convenient
+        self._steps_per_count = {}  # axis name (str) -> float
+        self._steps_per_meter = {}  # axis name (str) -> float
+        self._counts_per_meter = {}  # axis name (str) -> float
+
+        # Parse axis parameters and create axis
+        axes_def = {}  # axis name -> Axis object
+        for axis_name, axis_par in axes.items():
+            if 'axis_number' in axis_par.keys():
+                axis_num = axis_par['axis_number']
+                if axis_num not in range(128):
+                    raise ValueError("Invalid axis number %s, needs to be 0 <= int <= 127." % axis_num)
+                elif axis_num in self._axis_map.values():
+                    axname = self._axis_map[axis_num]
+                    raise ValueError("Invalid axis number %s, already assigned to axis %s." % (axis_num, axname))
+                else:
+                    self._axis_map[axis_name] = axis_par['axis_number']
+            else:
+                raise ValueError("Axis %s has no axis number." % axis_name)
+
+            if 'closed_loop' in axis_par.keys():
+                closed_loop = axis_par['closed_loop']
+            else:
+                closed_loop = False
+                logging.info("Axis mode (closed/open loop) not specified for axis %s. Assuming closed loop.", axis_name)
+            self._closed_loop[axis_name] = closed_loop
+
+            if 'encoder_resolution' in axis_par.keys():
+                self._counts_per_meter[axis_name] = 1 / axis_par['encoder_resolution']  # approximately 5e-6 m / step
+            else:
+                self._counts_per_meter[axis_name] = DEFAULT_COUNTS_PER_METER
+                logging.info("Axis %s has no encoder resolution, assuming %s." % (axis_name, 1 / DEFAULT_COUNTS_PER_METER))
+
+            if 'limit_type' in axis_par.keys():
+                limit_type = axis_par['limit_type']
+            else:
+                logging.info("Axis %s has not limit switch." % axis_name)
+                limit_type = 0
+
+            if 'range' in axis_par.keys():
+                axis_range = axis_par['range']
+            else:
+                axis_range = (0, STROKE_RANGE)
+                logging.info("Axis %s has no range. Assuming %s", axis_name, axis_range)
+
+            if 'spc' in axis_par.keys():
+                self._steps_per_count[axis_name] = axis_par['spc']  # approximately 5e-6 m / step
+                self._steps_per_meter[axis_name] = axis_par['spc'] * self._counts_per_meter[axis_name]
```

Comment:
My idea was to still have the parameter in the yaml file in case we want to override the value in flash memory. Maybe it's not necessary. What do you think?

Reply:
I'm only talking of the computation of `_steps_per_meter`. The "spc" parameter is fine.

Reply:
Ok, but then we need to have this code here. Later, I check if _steps_per_count is None and if so, I read the value from flash. If it's already set, I don't read from flash.

Reply:
As said in the previous comment 😉... I'm only talking of the computation of _steps_per_meter .

Reply:
ah, sorry, I didn't read carefully :)